msal/application.py

```diff
@@ -200,12 +206,19 @@ def __init__(
             or an X509 certificate container in this form::
 
                 {
-                    "private_key": "...-----BEGIN PRIVATE KEY-----...",
+                    "private_key": "...-----BEGIN PRIVATE KEY-----... in PEM format",
                     "thumbprint": "A1B2C3D4E5F6...",
                     "public_certificate": "...-----BEGIN CERTIFICATE-----... (Optional. See below.)",
                     "passphrase": "Passphrase if the private_key is encrypted (Optional. Added in version 1.6.0)",
                 }
 
+            MSAL Python requires a "private_key" in PEM format.
+            If your cert is in a PKCS12 (.pfx) format, you can also
+            `convert it to PEM and get the thumbprint <https://github.com/Azure/azure-sdk-for-python/blob/07d10639d7e47f4852eaeb74aef5d569db499d6e/sdk/identity/azure-identity/azure/identity/_credentials/certificate.py#L101-L123>`_.
+
+            The thumbprint is available in your app's registration in Azure Portal.
+            Alternatively, you can `calculate the thumbprint <https://github.com/Azure/azure-sdk-for-python/blob/07d10639d7e47f4852eaeb74aef5d569db499d6e/sdk/identity/azure-identity/azure/identity/_credentials/certificate.py#L94-L97>`_.
```

Comment:
It may not be a good idea to rely on downstream a library `azure-identity`'s code.

The code from https://github.com/Azure/azure-sdk-for-python/blob/07d10639d7e47f4852eaeb74aef5d569db499d6e/sdk/identity/azure-identity/azure/identity/_credentials/certificate.py#L94-L97 is not accurate as it passes `private_key` as `bytes`, not `str` (https://github.com/Azure/azure-sdk-for-python/issues/36578). This is against the MSAL's document.

Reply:
Thanks for your investigation!

Back then, we did not mean to "rely on a downstream library's code"; we just repurpose a code snippet which happens to be inside a downstream library.

Feel free to create a PR for the fix, preferably based on `openssl` command lines.

Reply:
> By the way, recent versions of MSAL started to support .pfx format and automatically calculate the thumbprint.

Maybe MSAL can also automatically calculate the thumbprint of a PEM certificate?